compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -599,7 +599,7 @@ fn process_rlink(sess: &Session, compiler: &interface::Compiler) {
 fn list_metadata(sess: &Session, metadata_loader: &dyn MetadataLoader) {
     match sess.io.input {
         Input::File(ref ifile) => {
```

Comment:
```suggestion
        Input::File(ref path) => {
```

and eliminate useless let-stmt below